.jenkins

```diff
@@ -470,12 +470,11 @@ pipeline {
                         sh '''rm -rf build && mkdir -p build && cd build && \
                               cmake \
                                 -DCMAKE_BUILD_TYPE=Release \
-                                -DCMAKE_CXX_STANDARD=14 \
+                                -DCMAKE_CXX_STANDARD=17 \
                                 -DCMAKE_CXX_FLAGS=-Werror \
                                 -DKokkos_ARCH_NATIVE=ON \
                                 -DKokkos_ENABLE_COMPILER_WARNINGS=ON \
-                                -DKokkos_ENABLE_DEPRECATED_CODE_3=ON \
-                                -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF \
+                                -DKokkos_ENABLE_DEPRECATED_CODE_3=OFF \
```

Comment:
I encountered some problems in the `partition_master` test but I didn't care much since we are going to remove deprecated code anyway.

Reply:
partition_master will not be removed until we have a replacement, its a critical dependency for a a very small number of important customers.

Reply:
```
[ RUN      ] openmp.partition_master
terminate called after throwing an instance of 'std::runtime_error'
terminate called after throwing an instance of 'std::runtime_error'
  what():  Kokkos::Impl::HostThreadTeamData::organize_pool ERROR pool already exists
  what():  Kokkos::Impl::HostThreadTeamData::organize_pool ERROR pool already exists
Aborted
```

Reply:
I still think that this is not a reason to hold the pull request since there are no code changes. I opened https://github.com/kokkos/kokkos/issues/5298 for this.

Reply:
Try adding `OMP_NESTED=TRUE` to the environment variables for the build. Some versions of gcc requires different environment variables to enable nested parallelism.

Reply:
Thanks, It's weird but it works.